### ISSUE TITLE
idea: passing Proc to unless

### ISSUE THREAD

**Issue by fwolfst** [xijo/restrict#3]
While trying to improve the explanation of `restrict, unless: , on: ` I was asking myself if a Proc/Lambda would not be more intent-revealing.

```ruby
restrict, unless: -> { manager_of?(load_item) }
```

No time for a PR, just an idea.

**Comment by xijo** [comment]
Good idea, reads pretty clean. No time for a PR as well though ¯\_(ツ)_/¯
